### ISSUE TITLE
(🐞) `poetry self update` installs poetry into a new location and breaks it with: "Poetry was not installed with the recommended installer"

### ISSUE THREAD

**Issue by KotlinIsland** [python-poetry/poetry#5032]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10
- **Poetry version**: 1.1.12 / 1.2.0a2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**:  N/A

## Issue

I installed poetry with the install-poetry.py script, and it installed into `\AppData\Roaming\Python\Scripts`.

But when I run `poetry self update` it installs poetry into a wacky different directory(`C:\Users\amongus\AppData\Local\Programs\Python\Python310\Scripts`) that doesn't let me run `poetry self update` any more:
`Poetry was not installed with the recommended installer, so it cannot be updated automatically.`
```
>where poetry
C:\Users\amongus\AppData\Roaming\Python\Scripts\poetry.exe

>poetry self update 1.1.11
Updating Poetry to 1.1.11

Updating dependencies
Resolving dependencies...

Package operations: 21 installs, 10 updates, 0 removals

  - Updating certifi (2021.5.30 -> 2021.10.8)
  - Updating charset-normalizer (2.0.6 -> 2.0.10)
  - Installing crashtest (0.3.1)
  - Updating idna (3.2 -> 3.3)
  - Installing pastel (0.2.1)
  - Installing pylev (1.4.0)
  - Updating urllib3 (1.26.7 -> 1.26.8)
  - Installing clikit (0.6.2)
  - Updating distlib (0.3.3 -> 0.3.4)
  - Updating filelock (3.3.0 -> 3.4.2)
  - Installing lockfile (0.12.2)
  - Installing msgpack (1.0.3)
  - Updating platformdirs (2.4.0 -> 2.4.1)
  - Installing ptyprocess (0.7.0)
  - Installing pywin32-ctypes (0.2.0)
  - Updating requests (2.26.0 -> 2.27.1)
  - Installing webencodings (0.5.1)
  - Installing cachecontrol (0.12.10)
  - Installing cachy (0.3.0)
  - Installing cleo (0.8.1)
  - Installing html5lib (1.1)
  - Installing keyring (21.8.0)
  - Updating packaging (21.3 -> 20.9)
  - Installing pexpect (4.8.0)
  - Installing pkginfo (1.8.2)
  - Installing poetry-core (1.0.7)
  - Installing requests-toolbelt (0.9.1)
  - Installing shellingham (1.4.0)
  - Installing tomlkit (0.8.0)
  - Updating virtualenv (20.8.1 -> 20.13.0)
  - Installing poetry (1.1.11)

Updating the poetry script

Poetry (1.1.11) is installed now. Great!

>where poetry
C:\Users\amongus\AppData\Local\Programs\Python\Python310\Scripts\poetry.exe
C:\Users\amongus\AppData\Roaming\Python\Scripts\poetry.exe

>poetry self update
  RuntimeError

  Poetry was not installed with the recommended installer. Cannot update automatically.

  at AppData\Local\Programs\Python\Python310\lib\site-packages\poetry\console\commands\self\update.py:389 in _check_reco
mmended_installation
      385│         current = Path(__file__)
      386│         try:
      387│             current.relative_to(self.home)
      388│         except ValueError:
    → 389│             raise RuntimeError(
      390│                 "Poetry was not installed with the recommended installer. "
      391│                 "Cannot update automatically."
      392│             )
      393│

```

**Comment by KotlinIsland** [comment]
This is a pretty serious bug that makes the install-poetry script practically useless IMO

**Comment by finswimmer** [comment]
I guess the root cause is the same as in https://github.com/python-poetry/poetry/issues/5046.

Could you please try my fix https://github.com/python-poetry/poetry/pull/5048, by installing it via:

```
 curl -sSL https://install.python-poetry.org | python - --git https://github.com/finswimmer/poetry@issue-5046-fix-self-update
```

Uninstall your current installation first.

fin swimmer

**Comment by m-dz** [comment]
I've run into a similar issue on Python 3.7.12, Win10 WSL2 (Ubuntu 20.04.2 LTS Focal Fossa), after running `poetry self update` on Poetry 1.1.11 and seeing "Updating Poetry to 1.1.12", `poetry --version` still gave 1.1.11. The aforementioned fix did the job.

**Comment by finswimmer** [comment]
Solved by #5048 and available since Poetry 1.1.13.
